src/idl_parser.cpp

```diff
@@ -555,7 +556,27 @@ CheckedError Parser::ParseType(Type &type) {
         return Error(
               "vector of union types not supported (wrap in table first).");
       }
-      type = Type(BASE_TYPE_VECTOR, subtype.struct_def, subtype.enum_def);
+      if (token_ == ':') {
+        if (!IsScalar(subtype.base_type)) {
+          return Error(
+                "arrays of non-scalar types not supported.");
+        }
+        NEXT();
+        if (token_ != kTokenIntegerConstant) {
```

Comment:
The current syntax is `field : [int:4]`, the brackets are parsed separately, as it is common part of arrays and vectors, but the difference is existing `:` in brackets, which I wrapped into `if` block.
There is a test (schema and generated file) for fixed length arrays, so you can see the result.